### PR TITLE
Magnetometer compensation based on throttle

### PR DESCRIPTION
**Pull request by RomanBapst**
This PR implements linear correction for each axis of a single magnetometer.
If there is a strong and linear relationship between the commanded throttle and magnetometer offset on a specific axis then this can be used to compensate for this effect. The model is really simple:

magX_corrected = mag_X_raw + throttle * d_magX_d_throttle
magY_corrected = mag_Y_raw + throttle * d_magY_d_throttle
magZ_corrected = mag_Z_raw + throttle * d_magZ_d_throttle

The correlation (linear model) needs to be identified offline.
If this proves to be useful then we will probably provide a python script which will identify the parameters based on a log file.
The compensation library introduced can be extended to use measured current instead of throttle.
For the future also a compensation based the individual ESC currents is thinkable.

### CODE REVIEW

**Comment by RomanBapst** [comment]
Currently I actually implemented a model that varies the offset using a cubic relationship to throttle.
I will add explanation for that.

File: src/lib/CMakeLists.txt [hunk 1]
```diff
@@ -59,3 +59,4 @@ add_subdirectory(terrain_estimation)
 add_subdirectory(tunes)
 add_subdirectory(version)
 add_subdirectory(WeatherVane)
+add_subdirectory(mag_compensation)
```
Comment (dagar) [inline src/lib/CMakeLists.txt]: We should really figure out some agreed upon naming conventions.

**Comment by dagar** [comment]
Looks like a good start, I'm interested to see the data.

As discussed one downside to doing this in sensors is that we can't adjust for ekf replay.

**Comment by RomanBapst** [comment]
Has been brought in by @baumanta in a separate PR, thanks!

**Pull request closed without merging**